workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+
+```py
+load_dotenv()
+app = Flask(__name__)
+
+if __name__ == '__main__':
+  app.run(host='0.0.0.0' , port = 9060)
+```
+
+The first one just finds and loads the your .env file and the second one says that this python file is the main file to run flask in. The third line is just doing a simple if check that will then run flask on localhost with the port being 9060.
+
+# Step 2: Getting Reddit API Key
+
+Getting your api key is very simple, go to the [Reddit Apps page](https://www.reddit.com/prefs/apps/) and click on `create another app...` at the bottom. ![The create another app button at the bottom](https://cloud-i2087nmyc.vercel.app/image.png)
+
+You should have a little section with `name`, `type`, `description`, `about url`, `redirect uri`. We only need the `name`, `type`, and `redirect uri`. For the name fill it with any name you want. For the type choose `script` and for the redirect uri, this doesn't matter, so just put `http://www.example.com/unused/redirect/uri`. ![The finished application creation](https://cloud-f67e393ze.vercel.app/image.png)
+
+In your repl.it open your .env file and make a variable called `SECRET` and another one called `ID`. Now go back to your reddit apps section and copy the secret from the secret part to your SECRET variable in your env file. Also copy the section just under your personal use script to your ID variable.
+
+![](https://cloud-119djxpws.vercel.app/image.png)
+
+We have the authentication setup which is easier done than said. Now one last step.
+
+We need to make a reddit variable that connects to the reddit api. Make a reddit = praw.Reddit() function. Now the next steps all go into the parenthesis. All it requires is three things, `client_id` which is your `"ID"` variable. So, do `client_id=os.getenv("ID"),`. The next one is client_secret. `client_secret=os.getenv("SECRET"),`. The final one is user_agent which is just a fancy name for what browser on what device. We will use a sample one as it doesn't actually matter which one you use. `user_agent="Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0")`
+
+![What the connection to Reddit should look like](https://cloud-ggp6iugz2.vercel.app/image.png)
+
+To test it out, run `print(reddit.read_only)`, if it returns true then you are golden.
+
+# Step 3: Setting up Flask Pages
+
+Make a folder called `templates` and inside of `templates` add an `index.html` file.
+
+![Your folder structure for the rest of this workshop](https://cloud-qvylewnf8.vercel.app/image.png)  
+Setup this index.html like any other html file with a header and a body.
+
+We want to load this index.html into our view. So, make an `<h1>` in the body that just says, `"Hello World"`. Now go to the main.py file and add an `@app.route('/')`. This means that the root (/) should load whatever function is under it. Now directly underneath the @app.route, add a function called front_page that returns render_template('index.html').
```

Comment:
Much more detailed steps!

Open the `index.html` file in your project directory and add:

```html
<h1>Hello world</h1>
```

Then, return to the `main.py` file. At the end of your file, add:

```py
@app.route(‘/‘)
??? I dont understand the rest of your steps but whatever this paragraph is describing, translate it to code form here
```

Reply:
Done!